src/checkpoints/checkpoints.cpp

```diff
@@ -301,8 +275,8 @@ namespace cryptonote
     if (m_db->get_immutable_checkpoint(&immutable_checkpoint, blockchain_height))
     {
       immutable_height = immutable_checkpoint.height;
-      if (rejected_by_service_node)
-        *rejected_by_service_node = (immutable_checkpoint.type == checkpoint_type::service_node);
+      if (service_node_checkpoint)
+        *service_node_checkpoint = (immutable_checkpoint.type == checkpoint_type::service_node);
     }
 
     m_oldest_allowable_alternative_block = std::max(immutable_height, m_oldest_allowable_alternative_block);
```

Comment:
This variable seems misnamed: it looks like it is storing the block just *before* the oldest allowable alt block (especially given that the next line uses `>` rather than `>=`).  Perhaps it should be something like `m_oldest_allowable_alt_block_base`.

Reply:
maybe just m_immutable_height? I don't fancy `alt_block_base` much either

Reply:
Yeah, that's a better name.